static/src/javascripts/projects/common/modules/commercial/user-features.ts

```diff
@@ -185,16 +184,16 @@ const userNeedsNewFeatureData = (): boolean =>
 	(adFreeDataIsPresent() && adFreeDataIsOld()) ||
 	(isDigitalSubscriber() && !adFreeDataIsPresent());
 
-const userHasDataAfterSignout = (): boolean =>
-	!isUserLoggedIn() && userHasData();
+const userHasDataAfterSignout = async (): Promise<boolean> =>
+	!(await isUserLoggedInOktaRefactor()) && userHasData();
 
 /**
  * Updates the user's data in a lazy fashion
  */
-const refresh = (): Promise<void> => {
-	if (isUserLoggedIn() && userNeedsNewFeatureData()) {
+const refresh = async (): Promise<void> => {
+	if ((await isUserLoggedInOktaRefactor()) && userNeedsNewFeatureData()) {
 		return requestNewData();
-	} else if (userHasDataAfterSignout() && !forcedAdFreeMode) {
+	} else if ((await userHasDataAfterSignout()) && !forcedAdFreeMode) {
 		deleteOldData();
 	}
 	return Promise.resolve();
```

Comment:
Is this return still needed? I think if you use `await` in a function without returning anything Typescript will consider the function to be a `Promise<void>`?!